ui/src/actions/planning.js

```diff
@@ -19,25 +19,56 @@
 
 export const PLAN_SET = 'PLAN_SET';
 
-function byDateAgeFn(a, b) {
-  // Order by start date.
-  // Replace "" with "Z" so the tasks with no starting date appear below.
-  var aStartDate = a.start_date || "Z";
-  var bStartDate = b.start_date || "Z";
-  if (aStartDate < bStartDate) {
-    return -1;
-  } else if (aStartDate > bStartDate) {
-    return 1;
+export function planSet(plan) {
+  // taskId -> parentTaskId
+  let parents = new Map();
+  for (var i = 0; i < plan.data.length; i++) {
+    parents.set(plan.data[i].id, plan.data[i].parent)
   }
 
-  // Order by task age.
-  var aTaskIDNumber = parseInt(a.url.split("T")[1])
-  var bTaskIDNumber = parseInt(b.url.split("T")[1])
-  return aTaskIDNumber - bTaskIDNumber;
-}
+  // taskId -> how many ancestor tasks it has
+  let levels = new Map();
+  function level(taskId) {
+    if (levels.has(taskId)) {
+      return levels.get(taskId);
+    }
+    let level = 0;
+    if (parents.has(taskId)) {
+      level = 1 + level(parents.get(taskId));
+    }
+    levels.set(taskId, level);
+    return level;
+  }
 
-export function planSet(plan) {
-  plan.data.sort(byDateAgeFn);
+  function byLevelDateAgeFn(a, b) {
+    // Order by number of ancestors.
+    let levelDelta = levels[a.id] - levels[b.id];
```

Comment:
It should be called here.  Surprising it works..

Reply:
"works"

Reply:
It doesn't.

Reply:
We agree it does not work!